hail/src/main/scala/is/hail/expr/types/physical/PDict.scala

```diff
@@ -16,7 +16,7 @@ abstract class PDict extends PContainer {
 
   def copy(keyType: PType = this.keyType, valueType: PType = this.valueType, required: Boolean = this.required): PDict
 
-  def elementType: PStruct = ???
+  def elementType: PStruct
```

Comment:
not sure if PBaseStruct is more appropriate here (from your comment), since PStruct is our new interface for implementations of PStruct, rather than PBaseStruct.